app/controllers/presentations_controller.rb

```diff
@@ -9,9 +9,15 @@ def new
   end
 
   def create
-    @presentation = Presentation.new(presentation_params).tap do |presentation|
-      presentation.person = current_person
+    if current_person.admin?
+      @presentation = Presentation.new(admin_presentation_params)
+    else
+      @presentation = Presentation.new(presentation_params)
+                                  .tap do |presentation|
```

Comment:
I dont think #tap is needed here. I believe you can pass a block directly to the #new method, which will yield the new instance.

Alternatively you might be able to the presentation params to add in current_person. 
`Presentation.new(person: current_person, **presentation_params)` 

Or you could also call merge on the presentation params hash 
`Presentation.new(presentation_params.merge(person: current_person))`